server/traces/traces.go

```diff
@@ -16,6 +16,50 @@ type Trace struct {
 	Flat     map[trace.SpanID]*Span `json:"-"`
 }
 
+func NewTrace(traceID string, spans ...Span) Trace {
```

Comment:
This can be named `New`, so the calls look like `traces.New(...)`

Reply:
Also, it looks easier to pass a `[]Span` than to expand an array on every call. I wouldn't expect it to be called with a manual list of `Span`s